zerver/actions/message_send.py

```diff
@@ -477,7 +477,8 @@ def do_schedule_messages(send_message_requests: Sequence[SendMessageRequest]) ->
         scheduled_message.recipient = send_request.message.recipient
         topic_name = send_request.message.topic_name()
         scheduled_message.set_topic_name(topic_name=topic_name)
-        scheduled_message.content = send_request.message.content
+        rendering_result = render_markdown(send_request.message, send_request.message.content, send_request.realm)
+        scheduled_message.content = rendering_result.rendered_content
```

Comment:
This is going to be problematic; the `rendered_content` needs to be a separate field.

(We need the original `content` to be stored for when the message gets actually sent?)

Reply:
Since we don't have existing UI/documentation for scheduled messages, we probably don't need any fancy migration code for ones without this feature, but I have checked and Zulip Cloud does have some with `delivered: True` created by folks playing with the API. We may end up wanting to have a migration delete `ScheduledMessage` objects with `delivered: true` on a one-time basis as part of future migrations.

As a sidenote, if we want to not just delete `ScheduledMessage` objects when a message is sent, should we change the database model to replace `delievered: True` with `delivered_message_id: {id}`, i.e. a reference to the actual message that was sent? Maybe a question for discussion about the data model for these in #backend. 

A related database model design question is whether we can have `ScheduledMessage` inherit from `AbstractMessage` or some new abstract base class; for example, maybe we rename `AbstractMessage` to `AbstractSentMessage`, and create a new abstract class `AbstractMessage` that `AbstractSentMessage` subclasses but that doesn't have fields like `edit_history` and `date_sent` that won't make sense for a `ScheduledMessage`.